client/src/components/export/functions.tsx

```diff
@@ -0,0 +1,190 @@
+        // Check if the user wants to export this field.
+        // If its an object, we need to recurse into it.
+        if (shouldExport !== true && !(value instanceof Object)) {
+            continue;
+        }
+
+        if (value instanceof Object) {
+            const subMap = csvItem(prefix + key + '.', value, opts[key]);
+            subMap.forEach((val, key) => {
+                result.set(key, val);
+            });
+        } else {
+            result.set(prefix + key, value);
+        }
+    }
+
+    return result;
+}
+
+// Recusively create a map of the items to be exported.
+function csvHeaderRecursion(prefix: string, opts: any): Map<string, string> {
+    const result = new Map<string, string>();
+
+    for (let key in opts) {
+        const shouldExport = opts[key];
+        // Check if the user wants to export this field.
+        // If its an object, we need to recurse into it.
+        if (shouldExport !== true && !(shouldExport instanceof Object)) {
+            continue;
+        }
+
+        const header = opts[key];
+
+        if (header instanceof Object) {
+            const subMap = csvHeaderRecursion(prefix + key + '.', header);
+            subMap.forEach((val, key) => {
+                result.set(key, val);
+            });
+        } else {
+            result.set(prefix + key, header);
+        }
+    }
+
+    return result;
+}
+
+// Sort the CSV entries by key.
+// This is to ensure that the values are always in the same order.
+// This also ensures that id is always the first column.
+function csvSort(aAny: [string, string], bAny: [string, string]) {
+    const a = aAny[0].toString();
+    const b = bAny[0].toString();
+
+    const remainingA = a.split('.').slice(0, -1).join('.');
+    const remainingB = b.split('.').slice(0, -1).join('.');
+
+    console.log(remainingA, remainingB);
+
+    const compare = remainingA.localeCompare(remainingB);
+    if (compare !== 0) {
+        return compare;
+    }
+
+    const lastBPart = b.split('.').pop();
+    const lastAPart = a.split('.').pop();
+
+    if (lastAPart === undefined || lastBPart === undefined) {
+        return 0;
+    }
+
+    if (lastAPart === 'id') {
+        return -1;
+    }
+
+    if (lastBPart === 'id') {
+        return 1;
+    }
+
+    return lastAPart.localeCompare(lastBPart);
+}
+
+/// Function that takes in an array of items and downloads the resulting QR code files.
+/// The title is used as the filename for the QR code.
+/// The data is the content of the QR code.
+export const exportQRCode = (qrCodeContents: {title: string, data: string}[], opts: QRExportOptions) => {
+    const apiEndpoint = import.meta.env.VITE_APIURL;
+
+    const url = `${apiEndpoint}/qr`;
+
+    for (let i = 0; i < qrCodeContents.length; i++) {
+        const content = qrCodeContents[i];
+
+        const body = JSON.stringify({
+            data: content.data,
+            box_size: opts.boxSize,
+            border: opts.padding
+        });
+
+        fetch(url, {
```

Comment:
You can return a the response form the API as an inline or attachment type.
Simply opening it would cause a download of the image, skipping the client side login.